orm_lib/src/SqlBinder.cc

```diff
@@ -266,22 +259,59 @@ SqlBinder &SqlBinder::operator<<(std::nullptr_t nullp)
     return *this;
 }
 
+SqlBinder &SqlBinder::operator<<(DefaultValue dv)
+{
+    (void)dv;
+    if (type_ == ClientType::PostgreSQL)
+    {
+        utils::replaceAll(*sqlPtr_,
```

Comment:
There's potential error in edge case with [dollar quoted constants][1]. `$0000$ INSERT INTO some_table (some_col) VALUES ($0); $0000$` would be incorrectly replaced into `default000$ INSERT INTO some_table (some_col) VALUES (default); default000$`

[1]: https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-DOLLAR-QUOTING

Reply:
Yes, this condition should be considered.

Reply:
@interfector18 @marty1885 
We can avoid replacing errors with regular expressions:
```regex
/$n([ ,\,\)])/default\1/g
```

Reply:
We could use regex for the first replacment, that replaces to default, but how would we do the second one? As the decrement replacement to $1 would for ex. replace 
`insert into x values(default, $2)` into 
`insert into x values(default, [ ,);])`.
I don't know of a way to escape that and there is the additional case of `where $1<=$2` and other comparison opetators.

We should use `\$n\b`. [Ex.](https://regex101.com/r/1b9xTZ/1)

I was hoping we could avoid regex but I guess it's the only way to correctly code this, short of writing a parser.

Reply:
We could use a loop for this, use regexs to find and replace `$n,$n+1, ...` to `default, $n, ...`

Reply:
> I was hoping we could avoid regex but I guess it's the only way to correctly code this, short of writing a parser.

I agree. Standard library's regex is far from the fastest thing in the world. But this shouldn't be the bottleneck. And still better than writing a parser. If that's a concern, I suggest writing the algorithm by hand.

Reply:
> We could use a loop for this, use regexs to find and replace `$n,$n+1, ...` to `default, $n, ...`

Yes, but $n is a special case for regex substitution, `$1` would substitute with whatever was found inside first `( )` or just leave it empty is there is no `( )` in search regex, I've found std::regex_constants::format_sed to resolve this issue.

Reply:
> Yes, but $n is a special case for regex substitution, $1 would substitute with whatever was found inside first ( ) or just leave it empty is there is no ( ) in search regex

I'm not sure if it's the same thing. I've discussed with @an-tao earlier that some specific adversarial input could make regex misbehave (Like putting a part of the query in the $..$ sequence). But we agreed that those cases can only happen if the user deliberately causes it. And we really can't help is that case.

Reply:
I mean for this snippet of code
```
std::string s =
      "$0000$ INSERT $0 INTO some_table (some_col) $0 VALUES ($0) WHERE "
      "$1<$2 and $2!=$3 and $4>=$6; $0000$";
  std::cout << "input string:\n" << s << "\n";

  std::regex r("\\$0\\b");
  s = std::regex_replace(s.data(), r, "default");

  size_t i = 1;
  while ((s.find("$" + std::to_string(i))) != std::string::npos) {
    r = "\\$" + std::to_string(i) + "\\b";
    s = std::regex_replace(s.data(), r,
                           std::string("$" + std::to_string(i - 1))/*, std::regex_constants::format_sed */);
    ++i;
  }

  cout << "output:\n" << s << "\n";
```

The output is:
```
❯ g++ -std=c++14 test.cpp -o test && ./test
input string:
$0000$ INSERT $0 INTO some_table (some_col) $0 VALUES ($0) WHERE $1<$2 and $2!=$3 and $4>=$6; $0000$
output:
$0000$ INSERT default INTO some_table (some_col) default VALUES (default) WHERE $1< and != and >=$6; $0000$
```

Using std::regex_constant::format_sed, fixes this issue.